pkgs/development/haskell-modules/configuration-hackage2nix.yaml

```diff
@@ -865,8 +865,8 @@ default-package-overrides:
   - haskell-gi-base ==0.23.0
   - haskell-gi-overloading ==1.0
   - haskell-lexer ==1.0.2
-  - haskell-lsp ==0.15.0.0
-  - haskell-lsp-types ==0.15.0.0
+  - haskell-lsp ==0.17.0.0
+  - haskell-lsp-types ==0.17.0.0
```

Comment:
In general, the Haskell stuff in nixpkgs tracks the latest LTS release.

If you look at the top of this list here, you can see that these packages are locked to versions from the latest LTS.  (peti updates these weekly, you can watch live if you're interested: https://www.twitch.tv/peti343)

It appears that the latest version of ghcide requires `haskell-lsp ==0.17.*, haskell-lsp-types ==0.17.*`, which is later than the version from LTS-14.

However, hackage2nix generates the latest versions of packages in the LTS as well, so `haskellPackages.haskell-lsp_0_18_0_0` and `haskellPackages.haskell-lsp-types_0_18_0_0` should be available.

I suggest trying to override ghcide to use these two package versions and then jailbreaking the cabal file.

Here's an example of `hnix` overriding package versions as well as using `doJailbreak`:

https://github.com/NixOS/nixpkgs/blob/63f66812bd263f96ec3c335b4160d8b2a2fe69ad/pkgs/development/haskell-modules/configuration-common.nix#L230-L233

Could you update this PR to remove the above lines from `configuration-hackage2nix.yaml`, and instead adding an entry for `ghcide` to `configuration-common.nix`?

Reply:
> I suggest trying to override ghcide to use these two package versions and then jailbreaking the cabal file.

Oh, and let me know if this doesn't work.  There is a way to force `hackage2nix` to generate arbitrary package versions (so we could make it generate `haskellPackages.haskell-lsp_0_17_0_0` for `ghcide`).

Reply:
Thanks for your explanations! Do I have to run `hackage2nix` locally to test my changes?

Reply:
@turion No, peti runs `hackage2nix` locally, so you don't have to do it in this PR.

Reply:
Unfortunately `ghcide` does not build with `haskell-lsp-0.18`. We probably need to make a `haskellPackages.haskell-lsp_0_17_0_0 `, but I don't know how. In parallel, I'll start and upgrade `ghcide` to build with a newer `haskell-lsp`.

Reply:
(This is known: https://github.com/digital-asset/ghcide/issues/219 It is fixed on `master`, but there has been no release since.)

Reply:
@turion thanks for the update.  I think there are two possibilities:

1. Get ghcide to release a new version that supports `haskell-lsp-0.18`, and then do as suggested above and send a PR to nixpkgs to override the version of `haskell-lsp` that the `ghcide` package uses.

2. Send a PR to nixpkgs that generates an older version of `haskell-lsp`.  You can do it by adding a line like https://github.com/NixOS/nixpkgs/blob/d0e512d6930490337a51f245f1827fa16ca01528/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L2510 for `haskell-lsp == 0.17.0.0`

    When this gets merged in (and `hackage-packages.nix` gets regenerated), there will be a new `haskellPackages.haskell-lsp_0_17_0_0` generated.  You can then use this in an override for `ghcide` as described above.

Solution 1 would be easier for us (and better for the community).  However, it is more work for you, and we have to rely on the `ghcide` maintainers merging your PR quickly and making a new release.  If the `ghcide` maintainers aren't quick with responding to your PR and you get tired of waiting, let's go with solution 2.

Reply:
I see. I'm working towards 1. then first :)

Reply:
`ghcide` with `haskell-lsp-0.18` is released. I don't really understand how this change gets here. Do I just wait and rebase?

Reply:
@turion

> I don't really understand how this change gets here.

I think there is an automatic script that updates the `haskell-updates` branch with recent package versions from Hackage.  (Or maybe it is semi-automatic and peti has to run it by hand)

You can see the current state of `haskell-updates` here (including a bunch of commits with the message `hackage-packages.nix: automatic Haskell package set update`):

https://github.com/NixOS/nixpkgs/pull/75429

At some point (maybe today or tomorrow?) one of these Haskell package set updates should contain the new version of `ghcide`.  At that point you should be able to rebase this PR on `haskell-updates` and use the new version of `haskell-lsp`.

Reply:
Ok. Thanks for all your explanations!

Is this whole workflow documented somewhere? Just in case peti goes on a 2 month vacation and someone needs to pick up the reign ;)

Reply:
It is somewhat documented in the `hackage2nix` Readme:

https://github.com/NixOS/cabal2nix/tree/master/hackage2nix

There is also a youtube video showing the steps:

https://www.youtube.com/watch?v=qX0mgtSm360

Also, peti streams live these updates, so if you watch him doing it once or twice, you should get a good idea of how to do it:

https://discourse.nixos.org/t/please-join-the-new-weekly-haskell-updates-video-livestream/4420